openapi/src/main/java/io/helidon/openapi/OpenAPISupport.java

```diff
@@ -108,63 +106,58 @@ public void update(final Routing.Rules rules) {
      * @param rules routing rules to be augmented with OpenAPI endpoint
      */
     public void configureEndpoint(Routing.Rules rules) {
-        try {
-            initializeOpenAPIDocument(openAPIConfig);
-        } catch (IOException ex) {
-            throw new RuntimeException("Error initializing OpenAPI information", ex);
-        }
 
         rules.get(JsonSupport.create())
                 .get(webContext, this::prepareResponse);
     }
 
     /**
-     * Prepares the information used to create the OpenAPI document for
-     * endpoints in this application.
+     * Prepares the OpenAPI model that later will be used to create the OpenAPI
+     * document for endpoints in this application.
      *
      * @param config {@code OpenApiConfig} object describing paths, servers,
      * etc.
-     * @throws IOException in case of errors reading any existing static OpenAPI
-     * document
+     * @return the OpenAPI model
+     * @throws RuntimeException in case of errors reading any existing static
+     * OpenAPI document
      */
-    void initializeOpenAPIDocument(final OpenApiConfig config) throws IOException {
-        try (OpenApiStaticFile staticFile = buildOpenAPIStaticFile()) {
+    private OpenAPI prepareModel(final OpenApiConfig config,
+            final IndexView indexView, OpenApiStaticFile staticFile) {
+        try {
             synchronized (OpenApiDocument.INSTANCE) {
                 OpenApiDocument.INSTANCE.reset();
                 OpenApiDocument.INSTANCE.config(config);
                 OpenApiDocument.INSTANCE.modelFromReader(OpenApiProcessor.modelFromReader(config, getContextClassLoader()));
                 OpenApiDocument.INSTANCE.modelFromStaticFile(OpenApiProcessor.modelFromStaticFile(staticFile));
                 if (isAnnotationProcessingEnabled(config)) {
-                    expandModelUsingAnnotations(config);
+                    expandModelUsingAnnotations(config, indexView);
                 } else {
                     LOGGER.log(Level.FINE, "OpenAPI Annotation processing is disabled");
                 }
                 OpenApiDocument.INSTANCE.filter(OpenApiProcessor.getFilter(config, getContextClassLoader()));
                 OpenApiDocument.INSTANCE.initialize();
-                model = OpenApiDocument.INSTANCE.get();
+                return OpenApiDocument.INSTANCE.get();
             }
+        } catch (IOException ex) {
+            throw new RuntimeException("Error initializing OpenAPI information", ex);
         }
     }
 
     private boolean isAnnotationProcessingEnabled(OpenApiConfig config) {
         return !config.scanDisable();
     }
 
-    private void expandModelUsingAnnotations(final OpenApiConfig config) throws IOException {
-        if (builder.indexView() != null) {
+    private void expandModelUsingAnnotations(final OpenApiConfig config, IndexView indexView) throws IOException {
```

Comment:
You should be consistent with the use of `final` for parameters. And I would be remiss if I didn't say that using it at all is controversial, and we have been encouraged not to.

Reply:
I removed all `final` from parameter declarations in `OpenAPISupport`.